migrations/scripts/generate_schema

```diff
@@ -0,0 +1,255 @@
+
+      File.open(output_filename, "w") { |f| f << @output_stream.string.chomp }
+    end
+
+    def generate_column(column)
+      @output_stream.print "  #{column.name} #{type(column)}"
+      @output_stream.print " NOT NULL" unless column.null
+      @output_stream.puts ","
+    end
+
+    def generate_index(table_name, index)
+      @output_stream.print "CREATE "
+      @output_stream.print "UNIQUE " if index[:unique]
+      @output_stream.print "INDEX #{index[:name]} ON #{table_name} (#{index[:columns].join(", ")})"
+      @output_stream.print " #{index[:condition]}" if index[:condition].present?
+      @output_stream.puts ";"
+    end
+
+    def column_list_for(table_name)
+      ignore_columns = @table_configs.dig(table_name, :ignore) || []
+      include_columns = @table_configs.dig(table_name, :include) || []
+
+      include_columns.present? ? [:include, include_columns] : [:ignore, ignore_columns]
+    end
+
+    def generate_table(name)
+      puts "Generating #{name}..."
+
+      column_records = columns(name)
+      mode, column_list = column_list_for(name)
+      indexes = indexes(name)
+      configured_primary_key = primary_key(name)
+
+      primary_key, composite_key =
+        if configured_primary_key.present?
+          [configured_primary_key].flatten.each do |pk|
+            if column_records.map(&:name).exclude?(pk)
+              raise "Column named '#{pk}' does not exist in table '#{name}'"
+            end
+          end
+
+          [
+            configured_primary_key,
+            configured_primary_key.is_a?(Array) && configured_primary_key.length > 1,
+          ]
+        else
+          virtual_table?(name) ? [] : [@core_db_connection.primary_key(name), false]
+        end
+
+      @output_stream.puts ""
+      @output_stream.puts "CREATE TABLE #{name} ("
+
+      if !composite_key && primary_key.present?
+        primary_key_column = column_records.find { |c| c.name == primary_key }
+
+        if (mode == :include && column_list.include?(primary_key_column.name)) ||
+             (mode == :ignore && column_list.exclude?(primary_key_column.name))
+          @output_stream.puts "  #{primary_key_column.name} #{type(primary_key_column)} NOT NULL PRIMARY KEY,"
+        end
+      end
+
+      column_records.each do |column|
+        next if @global_column_ignore_list.include?(column.name)
+        next if (mode == :ignore) && column_list.include?(column.name)
+        if !column.is_a?(CustomColumn) && (mode == :include) && column_list.exclude?(column.name)
+          @indirectly_ignored_columns[name] << column
+
+          next
+        end
+        next if !composite_key && (column.name == primary_key)
+
+        generate_column(column)
+      end
+
+      @output_stream.puts "  PRIMARY KEY (#{primary_key.join(", ")})," if composite_key
+      @output_stream.seek(@output_stream.pos - 2) # Go back two chars, helps remove last new line & trailing comma
+      @output_stream.puts "\n);"
```

Comment:
I considered that, but i figured moving the pointer back was much faster than initializing and joining a new array for each table. 

This approach is understandably  super coupled with the details of the `generate_column` method but I'm comfortable with this approach for now since it's use is limited to just this one location.